core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala

```diff
@@ -80,6 +81,19 @@ abstract class IntegrationTestHarness extends KafkaServerTestHarness {
     }
   }
 
+  private def insertControllerListenersIfNeeded(props: Seq[Properties]): Unit = {
+    if (isKRaftTest()) {
+      props.foreach { config =>
+        // Add a security protocol for the CONTROLLER endpoint, if one is not already set.
+        val securityPairs = config.getProperty(KafkaConfig.ListenerSecurityProtocolMapProp, "").split(",")
+        if (!securityPairs.exists(_.startsWith("CONTROLLER:"))) {
```

Comment:
nit: maybe we could pull the controller listener name from the config to make this logic a bit more general

Reply:
ok